### PR TITLE
No throw

### PR DESCRIPTION
**Pull request by pselle**
- also fixes issue with `custom_metrics = false` on old agents

### CODE REVIEW

**Comment by pselle** [comment]
@ewindisch https://github.com/iopipe/iopipe-payload/pull/29/commits/e19501810581478562840876e7d8587663724e03 maybe you only looked at the first commit and not the whole diff?

File: util/validateTypes.js [hunk 1]
```diff
@@ -13,7 +14,8 @@ function validateInteger(item, prop) {
     if (!isNaN(Number(item))) {
       return Math.trunc(Number(item))
     }
-    throw new TypeError(`${item} is not a number for ${prop}`)
+    console.log(`${item} is not a number for ${prop}`)
```
Comment (ewindisch) [inline util/validateTypes.js]: Lets kill these prints?

Comment (pselle) [inline util/validateTypes.js]: @ewindisch imo I think we should log, otherwise we won't have any logs when data is dropped

Comment (ewindisch) [inline util/validateTypes.js]: I don't think libraries should ever print or log (except in functions explicitly for that purpose). Maybe we can return "err, result" and check those on consumption?

Comment (pselle) [inline util/validateTypes.js]: Hmm, I thought we didn't want to `try {} catch` hence this approach, so if we _do_ want to continue to have this library throw, that's ok

Comment (ewindisch) [inline util/validateTypes.js]: We can use return values instead of raise/try/catch?

Comment (pselle) [inline util/validateTypes.js]: We can't return a value of a different type (ex an error message cannot go into a number field). Sending back an (err, result) is probably the best approach here

Comment (pselle) [inline util/validateTypes.js]: It does however probably necessitate a complete re-working of how this library works though. We can chat on the call about it.

**Comment by pselle** [comment]
Closing this and opening another PR with the `custom_metrics = false` fix

**Pull request closed without merging**